apps/autoscheduler/CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+target_include_directories(retrain_cost_model_process
+                           PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../support)
+target_link_libraries(retrain_cost_model_process
+                      PRIVATE cost_model train_cost_model)
+add_dependencies(retrain_cost_model_process cost_model train_cost_model)
+
+# =======================================================
+
+# libauto_schedule
+add_library(auto_schedule
+            SHARED
+            ASLog.cpp
+            AutoSchedule.cpp
+            DefaultCostModel.cpp
+            FunctionDAG.cpp
+            Weights.cpp
+            ${WF_CPP})
+
+if(HALIDE_LIBRARY_TYPE STREQUAL STATIC)
+  target_link_libraries(auto_schedule PRIVATE cost_model train_cost_model)
+else()
+  target_link_libraries(auto_schedule
+                        PRIVATE cost_model train_cost_model Halide)
+endif()
+
+if(NOT MSVC)
+  set_target_properties(auto_schedule PROPERTIES LINK_FLAGS "-rdynamic")
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+  set_target_properties(auto_schedule
+                        PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+endif()
+
+if(NOT HALIDE_ENABLE_RTTI)
+  if(NOT MSVC)
+    target_compile_options(auto_schedule PUBLIC "-fno-rtti")
+  else()
+    target_compile_options(auto_schedule PUBLIC "/GR-")
+  endif()
+endif()
+
+add_dependencies(auto_schedule cost_model train_cost_model)
+install(TARGETS auto_schedule LIBRARY DESTINATION bin)
+
+# =======================================================
+
+# demo_apps_autoscheduler
+halide_generator(demo.generator SRCS demo_generator.cpp)
+halide_library_from_generator(demo
+                              FUNCTION_NAME
+                              demo
+                              GENERATOR
+                              demo.generator
+                              GENERATOR_ARGS
+                              auto_schedule=true
+                              -p
+                              $<TARGET_FILE:auto_schedule>)
+add_dependencies(demo auto_schedule)
+
+_halide_genfiles_dir(demo GENFILES_DIR)
+set(DEMO_REGISTRATION_FILE ${GENFILES_DIR}/demo.registration.cpp)
+
+add_executable(demo_apps_autoscheduler ${HALIDE_TOOLS_DIR}/RunGenMain.cpp
+               ${DEMO_REGISTRATION_FILE})
+target_include_directories(
+  demo_apps_autoscheduler
+  PRIVATE "${HALIDE_INCLUDE_DIR}" "${HALIDE_TOOLS_DIR}")
+halide_use_image_io(demo_apps_autoscheduler)
+target_link_libraries(demo_apps_autoscheduler PRIVATE demo)
+add_dependencies(demo_apps_autoscheduler demo auto_schedule)
+
+add_custom_command(TARGET demo_apps_autoscheduler POST_BUILD
+                   COMMAND demo_apps_autoscheduler
+                           --benchmarks=all
+                           --benchmark_min_time=1
+                           --estimate_all
+                   WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+
+# =======================================================
+
+# included_schedule_file
+halide_generator(included_schedule_file.generator SRCS
```

Comment:
Nit: for generators that only get executed to form a single output, it's terser and cleaner to use a single `halide_library` rule, rather than separate `halide_generator` + `halide_library_from_generator` rules